### PR TITLE
Re-implement MM1 style damage

### PR DESCRIPTION
**Pull request by Wanted9811**

### CODE REVIEW

File: src/modules/gfx/packet.cpp [hunk 1]
```diff
@@ -85,6 +85,13 @@ void gfxPacket::GetTri(int* out, int index)
     out[2] = Indices[(3 * index) + 2];
 }
 
+void gfxPacket::SetTri(int* tri, int index)
```
Comment (Dummiesman) [inline src/modules/gfx/packet.cpp]: Could you make this function signature
(int i0, int i1, int i2, int index) instead?

File: src/modules/effects/damage.cpp [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+						currentVertex++;
+					}
+
+					packet->DoUnlock();
+					packet = packet->GetNext();
+				}
+			}
+		}
+
+		Reset(0);
```
Comment (Dummiesman) [inline src/modules/effects/damage.cpp]: Since this is a boolean, change this to false for clarity

**Review by Dummiesman — CHANGES_REQUESTED** [review]

File: src/modules/gfx/packet.cpp (hunk 1 above)
Comment (Wanted9811) [reply src/modules/gfx/packet.cpp]: Signature improved!

File: src/modules/effects/damage.cpp (hunk 2 above)
Comment (Wanted9811) [reply src/modules/effects/damage.cpp]: Done!

**Pull request merged**